packages/protocol/contracts/layer1/preconf/iface/IPreconfWhitelist.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+/// @title IPreconfWhitelist
+/// @custom:security-contact security@taiko.xyz
+interface IPreconfWhitelist {
+    /// @notice Emitted when a new operator is added to the whitelist.
+    /// @param operator The address of the operator that was added.
+    event OperatorAdded(address indexed operator);
+
+    /// @notice Emitted when an operator is removed from the whitelist.
+    /// @param operator The address of the operator that was removed.
+    event OperatorRemoved(address indexed operator);
+
+    error InvalidOperatorIndex();
+    error InvalidOperatorCount();
+
+    /// @notice Adds a new operator to the whitelist.
+    /// @param _operatorAddress The address of the operator to be added.
+    /// @dev Only callable by the owner or an authorized address.
+    function addOperator(address _operatorAddress) external;
```

Comment:
In the context of preconfirmation where blocks are constructed and shared within P2P network, to ensure these blocks originate from the current preconfirmer, the preconfirmer must sign the block's header. However, if the operator is a contract address, this would require a network operation to perform an `IERC1271` `isValidSignature` check. 

To optimize this process, it would be ideal for the operator to be represented as a pair of addresses `(preconfer, blockSigner)`, where:
- `preconfer` is the preconfirmer address responsible for calling `proposePreconfedBlocks`.
- `blockSigner` is the address used to sign block headers for preconfirmed blocks.

Let me know if you have alternative suggestions or additional thoughts!

Reply:
I initially considered modifying the protocol to include a `blockSigner` field in the metadata. However, this approach wouldn't work because the block is proposed *after* the shared block is validated in the P2P network. This means the `blockSigner` address must be known and available *early*, not just when a block or batch is proposed.

Reply:
> However, if the operator is a contract address, this would require a network operation to perform an IERC1271 isValidSignature check.

In what case would the operator want to be a contract address?